File: Invest-parent/Invest-Invest/Invest-Invest-Repository/src/main/java/com/illunex/invest/invest/entity/ExamMenu.java
package com.illunex.invest.invest.entity;

import com.illunex.invest.invest.entity.enumable.ViewMode;

import javax.persistence.*;
import java.util.List;

@Entity
public class ExamMenu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long examMenuIdx;
    Long userIdx;
    String title;
    ViewMode viewMode;

    @OneToMany
    List<ExamItemTemplate> examItemTemplate;
}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/ExamMenu.kt
package com.illunex.invest.invest.persistence.entity

import com.illunex.invest.invest.persistence.entity.enumable.ViewMode
import javax.persistence.*

@Entity
class ExamMenu {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var examMenuIdx: Long? = null
    var userIdx: Long? = null
    var title: String? = null
    var viewMode: ViewMode? = null

    @OneToMany
    var examItemTemplate: List<ExamItemTemplate>? = null
}
